megamek/src/megamek/client/ui/swing/boardview/BoardView.java

```diff
@@ -1126,7 +1126,7 @@ private synchronized void drawHexSpritesForHex(Coords c, Graphics g,
             Coords cp = sprite.getPosition();
             // This can potentially be an expensive operation
             Rectangle spriteBounds = sprite.getBounds();
-            if (cp.equals(c) && view.intersects(spriteBounds) && !sprite.isHidden()) {
+            if (cp != null && cp.equals(c) && view.intersects(spriteBounds) && !sprite.isHidden()) {
```

Comment:
I'd suggest moving the null check higher up and doing a continue before `sprite.getBounds()`


```java
if (cp == null) { 
  continue;
}

...
```

Reply:
That works; implemented.